### ISSUE TITLE
Error: Could not open URL 'banggood://m.banggood.com/onecentsnatch-index.html?utm_content=bgtr&p=7P091414086402201709&link_click_id=693742714772858139': No Activity found to handle Intent { act=android.intent.action.VIEW dat=banggood: flg=0x10000000 }]

### ISSUE THREAD

**Issue by knightkhan** [react-native-community/react-native-webview#1209]
unable to open link this massage appears any solution og it 
[Unhandled promise rejection: Error: Could not open URL 'banggood://m.banggood.com/onecentsnatch-index.html?utm_content=bgtr&p=7P091414086402201709&link_click_id=693742714772858139': No Activity found to handle Intent { act=android.intent.action.VIEW dat=banggood: flg=0x10000000 }]
`import React, {Component} from 'react';
import {Button, Text, View} from 'react-native';

import WebView from 'react-native-webview';

export default class Scrolling extends Component {

  render() {
    return (
      <View>
        <View style={{ height: 120 }}>
          <WebView
            source={{uri: 'https://adhitzads.com/1082501'}}
            automaticallyAdjustContentInsets={true}
            javaScriptEnabled={true}
            originWhitelist={['https://*', 'git://*']}
          />
        </View>
      </View>
    );
  }
}`

**Comment by bhadreshg-nimblechapps** [comment]
Please check this
https://github.com/react-native-community/react-native-webview/issues/1381#issuecomment-661644583

**Comment by bhadreshg-nimblechapps** [comment]
I have added some code for temporary solution.
Open WebViewShared.js file
Add below code
```
const arrayUrl = url.split(';');
var getUrl;
if (arrayUrl.length > 0) {
    const index = arrayUrl.findIndex((url) => url.includes('browser_fallback_url'));
    if (index > 0) {
        getUrl = arrayUrl[index].split('=')[1];
        getUrl = decodeURIComponent(getUrl);
        if (getUrl) {
            url = getUrl;
        }
    }
}
```
In createOnShouldStartLoadWithRequest variable return function after this
```
var url = nativeEvent.url, lockIdentifier = nativeEvent.lockIdentifier;
```

**Comment by bhadresh12** [comment]
I have added some code for solution.
Open WebViewShared.js file
```
const arrayUrl = url.split(';');
var getUrl;
if (arrayUrl.length > 0) {
    const index = arrayUrl.findIndex((url) => url.includes('browser_fallback_url'));
    if (index > 0) {
        getUrl = arrayUrl[index].split('=')[1];
        getUrl = decodeURIComponent(getUrl);
        if (getUrl) {
            url = getUrl;
        }
    }
}
```
In createOnShouldStartLoadWithRequest return function add above code after this
```
var url = nativeEvent.url, lockIdentifier = nativeEvent.lockIdentifier;
```

**Comment by Miti063** [comment]
This is fixed from version 11.0.0, but it breaks the hash changes (webview doesn't load URL if it contains a hash (#) after a 302 redirect). So if your app requires both the feature then you can install "10.8.0" version & apply this patch to support both.

 ```
diff --git a/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java b/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java
index 0461a0c..f9d0219 100644
--- a/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java
+++ b/node_modules/react-native-webview/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java
@@ -14,6 +14,7 @@ import android.net.http.SslError;
 import android.net.Uri;
 import android.os.Build;
 import android.os.Environment;
+import android.os.Message;
 import androidx.annotation.RequiresApi;
 import androidx.core.content.ContextCompat;
 import android.text.TextUtils;
@@ -181,6 +182,7 @@ public class RNCWebViewManager extends SimpleViewManager<WebView> {
     settings.setBuiltInZoomControls(true);
     settings.setDisplayZoomControls(false);
     settings.setDomStorageEnabled(true);
+    settings.setSupportMultipleWindows(true);
 
     settings.setAllowFileAccess(false);
     settings.setAllowContentAccess(false);
@@ -246,6 +248,11 @@ public class RNCWebViewManager extends SimpleViewManager<WebView> {
     view.getSettings().setJavaScriptEnabled(enabled);
   }
 
+  @ReactProp(name = "setSupportMultipleWindows")
+  public void setSupportMultipleWindows(WebView view, boolean enabled){
+    view.getSettings().setSupportMultipleWindows(enabled);
+  }
+
   @ReactProp(name = "showsHorizontalScrollIndicator")
   public void setShowsHorizontalScrollIndicator(WebView view, boolean enabled) {
     view.setHorizontalScrollBarEnabled(enabled);
@@ -1014,6 +1021,17 @@ public class RNCWebViewManager extends SimpleViewManager<WebView> {
       this.mWebView = webView;
     }
 
+    @Override
+    public boolean onCreateWindow(WebView view, boolean isDialog, boolean isUserGesture, Message resultMsg) {
+
+      final WebView newWebView = new WebView(view.getContext());
+      final WebView.WebViewTransport transport = (WebView.WebViewTransport) resultMsg.obj;
+      transport.setWebView(newWebView);
+      resultMsg.sendToTarget();
+
+      return true;
+    }
+
     @Override
     public boolean onConsoleMessage(ConsoleMessage message) {
       if (ReactBuildConfig.DEBUG) {

```
